server/db/dbManager.ts

```diff
@@ -24,23 +24,104 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
-		let posts;
-		if (parent) {
-			posts = await this.model.find({ parent }).sort({ upvotes: -1 }).lean().exec();
-			if (depth < parseInt(process.env.MAX_COMMENT_DEPTH)) {
-				for (const post of posts) {
-					post.children = [];
-					if (depth > 0) {
-						parentObject.children.push(post);
+	async getAllPosts(recursive: boolean, parent?: mongoose.Types.ObjectId, parentObject = { children: [] }): Promise<object> {
+		if (parent && recursive) {
```

Comment:
Maybe, it worth renaming the `recursive` variable to more understanding naming that explains the output result. As example `returnWithComments` or just `withComments`.